--- /home/akashk4/tensor/tlx/llvm-project/llvm/./include/llvm/InitializePasses.h	2022-04-13 11:35:53.183647365 -0500
+++ ./include/llvm/InitializePasses.h	2022-04-13 09:57:35.763723585 -0500
@@ -60,6 +60,10 @@
 /// Initialize all passes linked into the CodeGen library.
 void initializeTarget(PassRegistry&);
 
+
+void initializeTensorInfoWrapperPassPass(PassRegistry&);
+void initializeLowerTensorIntrinsicsLegacyPassPass(PassRegistry&);
+
 void initializeAAEvalLegacyPassPass(PassRegistry&);
 void initializeAAResultsWrapperPassPass(PassRegistry&);
 void initializeADCELegacyPassPass(PassRegistry&);
