--- /home/akashk4/tensor/tlx/llvm-project/llvm/./include/llvm/LinkAllPasses.h	2022-04-13 11:35:53.219650368 -0500
+++ ./include/llvm/LinkAllPasses.h	2022-04-13 09:57:35.767723921 -0500
@@ -233,6 +233,7 @@
       (void) llvm::createUnifyLoopExitsPass();
       (void) llvm::createFixIrreduciblePass();
       (void)llvm::createFunctionSpecializationPass();
+      (void) llvm::createLowerTensorIntrinsicsPass();
 
       (void)new llvm::IntervalPartition();
       (void)new llvm::ScalarEvolutionWrapperPass();
