--- /home/akashk4/tensor/tlx/llvm-project/llvm/./include/llvm/ADT/APFloat.h	2022-04-13 11:35:50.745444012 -0500
+++ ./include/llvm/ADT/APFloat.h	2022-04-13 09:57:35.744721988 -0500
@@ -156,7 +156,8 @@
     S_IEEEdouble,
     S_x87DoubleExtended,
     S_IEEEquad,
-    S_PPCDoubleDouble
+    S_PPCDoubleDouble,
+    S_CustomFP
   };
 
   static const llvm::fltSemantics &EnumToSemantics(Semantics S);
@@ -169,6 +170,7 @@
   static const fltSemantics &IEEEquad() LLVM_READNONE;
   static const fltSemantics &PPCDoubleDouble() LLVM_READNONE;
   static const fltSemantics &x87DoubleExtended() LLVM_READNONE;
+  static const fltSemantics &CustomFP() LLVM_READNONE;
 
   /// A Pseudo fltsemantic used to construct APFloats that cannot conflict with
   /// anything real.
@@ -1344,3 +1346,4 @@
 
 #undef APFLOAT_DISPATCH_ON_SEMANTICS
 #endif // LLVM_ADT_APFLOAT_H
+
