--- /home/akashk4/tensor/tlx/llvm-project/llvm/./lib/Analysis/Analysis.cpp	2021-11-19 17:32:37.761993212 -0600
+++ ./lib/Analysis/Analysis.cpp	2021-11-19 17:27:38.461919065 -0600
@@ -86,6 +86,7 @@
   initializeLCSSAVerificationPassPass(Registry);
   initializeMemorySSAWrapperPassPass(Registry);
   initializeMemorySSAPrinterLegacyPassPass(Registry);
+  initializeTensorInfoWrapperPassPass(Registry);
 }
 
 void LLVMInitializeAnalysis(LLVMPassRegistryRef R) {
