--- /home/akashk4/tensor/tlx/llvm-project/llvm/./lib/Transforms/IPO/PassManagerBuilder.cpp	2022-04-13 11:36:06.046720196 -0500
+++ ./lib/Transforms/IPO/PassManagerBuilder.cpp	2022-04-13 09:57:35.818728207 -0500
@@ -910,6 +910,7 @@
     // analysis to establish no-aliasing between loads and stores of different
     // columns of the same matrix.
     MPM.add(createEarlyCSEPass(false));
+    MPM.add(createLowerTensorIntrinsicsPass());
   }
 
   addExtensionsToPM(EP_VectorizerStart, MPM);
