--- /home/akashk4/tensor/tlx/llvm-project/llvm/./lib/Transforms/Scalar/CMakeLists.txt	2022-04-13 11:36:06.324743384 -0500
+++ ./lib/Transforms/Scalar/CMakeLists.txt	2022-04-13 09:57:35.823728628 -0500
@@ -52,6 +52,7 @@
   LowerExpectIntrinsic.cpp
   LowerGuardIntrinsic.cpp
   LowerMatrixIntrinsics.cpp
+  LowerTensorIntrinsics.cpp
   LowerWidenableCondition.cpp
   MakeGuardsExplicit.cpp
   MemCpyOptimizer.cpp
