--- /home/akashk4/tensor/tlx/llvm-project/llvm/./lib/Transforms/Scalar/CMakeLists.txt	2021-11-19 17:32:46.439720137 -0600
+++ ./lib/Transforms/Scalar/CMakeLists.txt	2021-11-19 17:27:38.489921411 -0600
@@ -52,6 +52,7 @@
   LowerExpectIntrinsic.cpp
   LowerGuardIntrinsic.cpp
   LowerMatrixIntrinsics.cpp
+  LowerTensorIntrinsics.cpp
   LowerWidenableCondition.cpp
   MakeGuardsExplicit.cpp
   MemCpyOptimizer.cpp
