--- /home/akashk4/tensor/tlx/llvm-project/llvm/./lib/Transforms/Scalar/Scalar.cpp	2021-11-19 17:32:46.633736390 -0600
+++ ./lib/Transforms/Scalar/Scalar.cpp	2021-11-19 17:27:38.493921747 -0600
@@ -116,6 +116,7 @@
   initializeLoopVersioningLegacyPassPass(Registry);
   initializeEntryExitInstrumenterPass(Registry);
   initializePostInlineEntryExitInstrumenterPass(Registry);
+  initializeLowerTensorIntrinsicsLegacyPassPass(Registry);
 }
 
 void LLVMAddLoopSimplifyCFGPass(LLVMPassManagerRef PM) {
