--- /home/akashk4/tensor/tlx/llvm-project/llvm/./lib/Transforms/Scalar/Scalar.cpp	2022-04-13 11:36:06.557762819 -0500
+++ ./lib/Transforms/Scalar/Scalar.cpp	2022-04-13 09:57:35.827728964 -0500
@@ -116,6 +116,7 @@
   initializeLoopVersioningLegacyPassPass(Registry);
   initializeEntryExitInstrumenterPass(Registry);
   initializePostInlineEntryExitInstrumenterPass(Registry);
+  initializeLowerTensorIntrinsicsLegacyPassPass(Registry);
 }
 
 void LLVMAddLoopSimplifyCFGPass(LLVMPassManagerRef PM) {
