--- /home/akashk4/tensor/tlx/llvm-project/llvm/./include/llvm/Transforms/Scalar.h	2022-04-13 11:35:54.583764140 -0500
+++ ./include/llvm/Transforms/Scalar.h	2022-04-13 09:57:35.772724341 -0500
@@ -562,6 +562,13 @@
 // and scatter intrinsics with scalar code when target doesn't support them.
 //
 FunctionPass *createScalarizeMaskedMemIntrinLegacyPass();
+
+
+//===----------------------------------------------------------------------===//
+
+FunctionPass *createLowerTensorIntrinsicsPass();
+
+
 } // End llvm namespace
 
 #endif
