--- /home/akashk4/tensor/tlx/llvm-project/llvm/./include/llvm/Transforms/Scalar.h	2021-11-19 17:32:37.378961126 -0600
+++ ./include/llvm/Transforms/Scalar.h	2021-11-19 17:27:38.454918479 -0600
@@ -562,6 +562,13 @@
 // and scatter intrinsics with scalar code when target doesn't support them.
 //
 FunctionPass *createScalarizeMaskedMemIntrinLegacyPass();
+
+
+//===----------------------------------------------------------------------===//
+
+FunctionPass *createLowerTensorIntrinsicsPass();
+
+
 } // End llvm namespace
 
 #endif
